src/test/ui/repeat-expr/infer.rs

```diff
@@ -0,0 +1,16 @@
+// check-pass
+
+#[derive(Clone, Default)]
+struct MaybeCopy<T>(T);
+
+impl Copy for MaybeCopy<u8> {}
+
+fn is_copy<T: Copy>(x: T) {
+    println!("{}", std::any::type_name::<T>());
+}
+
+fn main() {
+    is_copy(MaybeCopy::default());
```

Comment:
Wait, I'm confused as to why this works. Is `Default: Copy`?

Reply:
Oh, wait. This is because `MabeCopy::default(): ~const`, right?

Reply:
This worked because there was only one type `T` for which `MaybeCopy<T>` could be `Copy`. If we also had `impl Copy for MaybeCopy<i8> {}` then this would stop working: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a4fa25d4154048a62eca08aedcceb090

Reply:
Ah! Wow! Hadn't realized that that's how it works.